### ISSUE TITLE
PiVPN Profile download not working in Firefox

### ISSUE THREAD

**Issue by TheNuttyP** [mitchellurgero/openrsd#42]
I am running OpenRSD v161.ed7e23b on Stretch.

I can use the PiVPN Profiles page to create a new account, but when I try to download the profile, the browser tried to download and then stops. This is in Firefox 67.0.4 64-bit. If I try to download in Chrome, it works as expected.

**Comment by mitchellurgero** [comment]
Thanks @TheNuttyP ! I can look into this for you, maybe come up with a fix, but it doesn't happen for me on my install. Can you open the developer console in browser, switch to the network tab, and then try the download again so we can get a look at the error code(s) here?

**Comment by TheNuttyP** [comment]
I wonder what is different in my install. It was just installed fresh.

I did as you asked. There is no error. The status is 200, even though no file downloaded.

**Comment by mitchellurgero** [comment]
Can you screen record the steps to reproduce? If it's not in the downloads folder, is it saving to a different location??

**Comment by mitchellurgero** [comment]
An alternative way to download, for now, is to click the filename in the profile list. This will bring up the file contents you can copy that out to an *.ovpn file for importing into OpenVPN GUI, or mobile clients.

**Comment by TheNuttyP** [comment]
The alternate will be helpful. Thank you.

I have done a screen recording and I'm uploading it to share now. I'll comment back with a link once its uploaded (my internet is not the fastest).

**Comment by TheNuttyP** [comment]
https://drive.google.com/file/d/1JN77NtIeGKFV0qZJlgwdK9eomwZ3dx-C/view

**Comment by mitchellurgero** [comment]
Not to sound like an ass, but I assume you checked the downloads button (top right, arrow pointing down)? It's blue in the video meaning a file was downloaded.

**Comment by TheNuttyP** [comment]
Not sounding like an ass at all. It was blue from a different download that I checked to make sure downloads were working in general. I did check the downloads section and nothing has downloaded from this.

**Comment by mitchellurgero** [comment]
Alright - cool cool. Just making sure. This is pretty strange for sure though. My next question is: Was pivpn installed as the same user the web server uses? (Example: the user pi)

**Comment by mitchellurgero** [comment]
Also, on the Pi itself, can you grab the server logs (`/var/log/apache2/error.log` for apache.)

**Comment by TheNuttyP** [comment]
Ahhh, there's something to go on...
Since I'm using Stretch, I'm running on lighttpd. It is running as 'www-data' and PiVPN was installed as 'pi'
The lighttpd log is full of "Permission denied' trying to access the ovpn file.

I double checked the download from Chrome...even though it downloaded a file, the file was blank.
[error.log](https://github.com/mitchellurgero/openrsd/files/3340668/error.log)

**Comment by mitchellurgero** [comment]
Yea, the web server running OpenRSD (until I can come up with a new mech for authentication as a PAM user) must be the same user using pivpn. 

Lighttpd: pi:pi
pivpn:  pi:pi
sudo nopasswd:  pi:pi

**Comment by mitchellurgero** [comment]
Change the config for Lighttpd to run as the user AND group of "pi"

**Comment by TheNuttyP** [comment]
Did I miss a step in the instructions? I see where the username/group change is mentioned for Apache, but it specifically says to not do this for Lighttpd...or did I read it wrong?

**Comment by mitchellurgero** [comment]
Did you do this line:
`echo "www-data ALL=(ALL) NOPASSWD: ALL" | sudo tee --append /etc/sudoers.d/010_pi-nopasswd`
And restart the lighttpd service?

**Comment by mitchellurgero** [comment]
oh add the php-fpm user to the pi group, that should give FPM read access to the files  (sudo is not used to read the file in the download script)

I think the command for this is just:
`sudo adduser www-data pi`

**Comment by mitchellurgero** [comment]
Restart php-fpm when done of course.

**Comment by TheNuttyP** [comment]
ok, I changed the user and group used by Lighttpd...I also changed the ownership of the html directory to pi:pi

I did do the line you referenced, and added www-data to the pi group. I've done a full restart of the pi, just to be safe.

Now I am getting error "503 - Service Not Available"

**Comment by mitchellurgero** [comment]
This is why I dislike lighttpd. Honestly, can you move to using apache instead? If not then what I would do is have lighttpd run as www-data, change the ownership of /var/www back and make sure www-data is in the group pi.

**Comment by TheNuttyP** [comment]
No problem. I tried to switch over to Apache and started getting a whole bunch of crap at the top of the page when it loads. I think I'm going to do a full wipe and start over.

The Apache instructions use php5 which is deprecated and no longer available in Stretch. Can I use the "apt install" line that is for Lighttpd (since that references php7) but just substitute Apache in place of Lighttpd?

**Comment by mitchellurgero** [comment]
yes that will work (minus php-fpm, use apaches module for php)

**Comment by TheNuttyP** [comment]
ok, let me reimage the card...I'll report back once its clean installed.

**Comment by mitchellurgero** [comment]
I made new install instructions and added a new recommended "Run as a service" bit to README.MD. Check it out let me know how it works.

**Comment by TheNuttyP** [comment]
Understood. I had already gotten PiVPN installed and was working on the old instructions. I can re-image again and start over if you want me to test the new instructions?

**Comment by mitchellurgero** [comment]
You can actually do the new instructions but replace the openrsd user with the pivpn user.

(also modify /etc/systemd/system/openrsd.service file to reflect the user)

**Comment by TheNuttyP** [comment]
Ok...to be fair to all, I started from scratch again and followed the new instructions:

Line 1: there are also a lot of other packages that apt installs that are supposed dependencies of these packages...which leads to:
Line 2: one of the dependencies that gets installed is Apache2, so this directory already exists and I get an error here.
Line 3: cd isn't a command that sudo can be used with, so it errors out. Works fine if I do the cd without sudo.

Everything else seems to go smooth and installs. Once installed, when I load the dashboard I have a "There was an error checking for OpenRSD updates" message. If I got to the Updates page, I get "There was an error processing the update request", but I pressed on.

Trying to install PiVPN as the openrsd user...I get to the point of the install where you select which user PiVPN is supposed to use, and openrsd ins't on the list. pi is the only user that is listed. I went with that and pressed on.

I am now back to the terminal, but it has been sitting on "Restarting services" for about 10 minutes now
